src/pygama/flow/file_db.py

```diff
@@ -0,0 +1,390 @@
+
+            for tb in tier_tables:
+                parse_arr = np.array(list(fm.parse(template)))
+                names = list(parse_arr[:, 1])
+                if len(names) > 0:
+                    keyword = names[0]
+                    args = {keyword: tb}
+                    table_name = template.format(**args)
+                else:
+                    table_name = template
+
+                col = ls(f[table_name])
+                if col not in columns:
+                    columns.append(col)
+                    col_idx.append(len(columns) - 1)
+                else:
+                    col_idx.append(columns.index(col))
+
+            return pd.Series(
+                {f"{tier}_tables": tier_tables, f"{tier}_col_idx": col_idx}
+            )
+
+        columns = []
+        for tier in self.tiers:
+            self.df[[f"{tier}_tables", f"{tier}_col_idx"]] = self.df.apply(
+                update_tables_cols, axis=1, tier=tier
+            )
+
+        self.columns = columns
+
+        if to_file is not None:
+            log.debug(f"Writing column names to '{to_file}'")
+            flattened = []
+            length = []
+            for i, col in enumerate(columns):
+                if i == 0:
+                    length.append(len(col))
+                else:
+                    length.append(length[i - 1] + len(col))
+                for c in col:
+                    flattened.append(c)
+            columns_vov = VectorOfVectors(
+                flattened_data=flattened, cumulative_length=length
+            )
+            sto = LH5Store()
+            sto.write_object(columns_vov, "unique_columns", to_file)
+
+        return columns
+
+    def from_disk(self, filename: str):
+        """
+        Fills the dataframe (and configuration dictionary) with the information
+        from a file created by :meth:`to_disk`.
+        """
+
+        sto = LH5Store()
+        cfg, _ = sto.read_object("config", filename)
+        self.set_config(json.loads(cfg.value.decode()))
+
+        self.df = pd.read_hdf(filename, key="dataframe")
+
+        vov, _ = sto.read_object("columns", filename)
+        # Convert back from VoV of UTF-8 bytestrings to a list of lists of strings
+        vov = list(vov)
+        columns = []
+        for ov in vov:
+            columns.append([v.decode("utf-8") for v in ov])
+        self.columns = columns
+
+    def to_disk(self, filename: str, wo_mode="write_safe"):
+        """Serializes database to disk.
+
+        Parameters
+        -----------
+        filename
+            output LH5 file name.
+        wo_mode
+            passed to :meth:`~.lgdo.lh5_store.write_object`.
+        """
+        log.debug(f"Writing database to {filename}")
+
+        sto = LH5Store()
+        sto.write_object(
+            Scalar(json.dumps(self.config)), "config", filename, wo_mode=wo_mode
+        )
+
+        if wo_mode in ["write_safe", "w", "overwrite_file", "of"]:
+            wo_mode = "a"
```

Comment:
Why is the writing mode being changed to append? 

Also, after this in the code, `"a"` is just used directly regardless of the value of `wo_mode`

Reply:
The idea is that if the file is created or overwritten in the first call to `write_object` a few lines above, all the rest must be appended. Other write modes that assume an existing file can be just passed to `write_object`. Does it make sense?

Now let me actually ass `wo_mode` in the subsequent calls, that's my fault.